src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -1644,9 +1643,7 @@ private void ClearCachedAnalysisDataIfAnalyzed(SyntaxReference declaration, ISym
             ComputeDeclarationsInNode(semanticModel, symbol, declaringReferenceSyntax, topmostNodeForAnalysis, builder, cancellationToken);
 
             var isPartialDeclAnalysis = analysisScope.FilterSpanOpt.HasValue && !analysisScope.ContainsSpan(topmostNodeForAnalysis.FullSpan);
-            var nodesToAnalyze = shouldExecuteSyntaxNodeActions ?
```

Comment:
more correct fix seems cache depending on declaration, shouldExecuteSyntaxNodeActions, analysisScope.FilterSpanOpt.

and remove analysisScope from parameter and accept TextSpan directly. 

but I wasn't 100% sure what is cache expiration policy and such so.. made simplest fix that fix the problem I am investigating right now.